### DISCUSSION TITLE
concepts question

### DISCUSSION THREAD

**Discussion by avnav0** [dagster-io/dagster#11517] [Q&A]
wondering if someone can point me towards where i can learn about passing arguments through an op to a resource.

thanks! loving this framework so far.

**Comment by clairelin135** [comment]
Hi @avnav88! Not sure if this is what you mean, but if you create a custom resource object you could do something like this to pass arguments from an op to a resource:

```
class MyDBConnection:
    def add_file(self, file):
        ...


@resource
def db_conn_resource(init_context):
    return MyDBConnection()


@op(required_resource_keys={"db_conn"})
def my_op(context):
    context.resources.db_conn.add_file(...)


@job(resource_defs={"db_conn": db_conn_resource})
def my_job():
    my_op()
```

**Reply by avnav0** [reply]
thanks!
